x-pack/libbeat/autodiscover/providers/aws/elb/provider.go

```diff
@@ -87,7 +87,7 @@ func internalBuilder(uuid uuid.UUID, bus bus.Bus, config *Config, fetcher fetche
 		return nil, err
 	}
 
-	return &Provider{
+	p := &Provider{
 		fetcher:   fetcher,
 		period:    config.Period,
```

Comment:
`fetcher` and `period` are only used to build the watcher, if we build the watcher now in the same builder there is no need to store them in the `Provider`.

Reply:
That's a great point. I'll do that refactor in a subsequent PR since I want to merge this to fix the build.